src/Microsoft.AspNetCore.Authentication/ClaimsTransformationHandler.cs

```diff
@@ -12,10 +13,12 @@ namespace Microsoft.AspNetCore.Authentication
     public class ClaimsTransformationHandler : IAuthenticationHandler
     {
         private readonly IClaimsTransformer _transform;
+        private readonly HttpContext _httpContext;
 
-        public ClaimsTransformationHandler(IClaimsTransformer transform)
+        public ClaimsTransformationHandler(IClaimsTransformer transform, HttpContext httpContext)
```

Comment:
I've added `HttpContext` instead of actual `ClaimsTransformationContext`, because `AuthenticateAsync` method below receives `AuthenticateContext` param with current `Principal` in it.
Therefore I left construction of `ClaimsTransformationContext` just before `TransformAsync` call. This way it gets most recent `Principal` from `AuthenticateContext`.